src/nfa/operations.cc

```diff
@@ -1188,18 +1186,12 @@ std::set<mata::Word> mata::nfa::Nfa::get_words(unsigned max_length) {
     unsigned cur_length = 0;
     while (!worklist.empty() && cur_length < max_length) {
         new_worklist.clear();
-        for (const auto& state_and_word : worklist) {
-            State s_from = state_and_word.first;
-            const mata::Word& word = state_and_word.second;
-            for (const SymbolPost& sp : delta[s_from]) {
+        for (const auto& [state_from, word]: worklist) {
+            for (const Move& move: delta[state_from].moves()) {
                 mata::Word new_word = word;
-                new_word.push_back(sp.symbol);
-                for (State s_to : sp.targets) {
-                    new_worklist.push_back({s_to, new_word});
-                    if (final.contains(s_to)) {
-                        result.insert(new_word);
-                    }
-                }
+                if (move.symbol < first_epsilon) { new_word.push_back(move.symbol); }
```

Comment:
`curr_length` is not correctly updated with epsilon transitions. So it should be updated here I think.
But then, you cannot have epsilon loops, otherwise this procedure will never finish.

Reply:
The current length (and the maximal length) here was (and still is) implemented as the number of transitions made, not the length of the words (which it should be, as per the function description). That is OK until you start working with epsilons.

We could, for example, remove epsilon transitions to solve this issue (or simply start the removal by checking that the computation of epsilon closure for a given state does not try to add the same state twice, otherwise perform the actual removal).

Reply:
This will require a larger refactorization of the epsilon removal function and the function to get words of a specific length. Therefore, I will remove these changes from this PR and rework this in another PR. The only changes kept in this PR are the changes introducing `get_word()` which works correctly even with epsilons.